backends/ref/ceed-ref-operator.c

```diff
@@ -515,19 +501,12 @@ static int CeedCompositeOperatorApply_Ref(CeedOperator op, CeedVector invec,
     CeedRequest *request) {
   int ierr;
   CeedInt numsub;
-  CeedOperator_Ref *impl;
   CeedOperator *suboperators;
   ierr = CeedOperatorGetNumSub(op, &numsub); CeedChk(ierr);
   ierr = CeedOperatorGetSubList(op, &suboperators); CeedChk(ierr);
 
-  // Overwrite outvec with first output
-  ierr = CeedOperatorApply(suboperators[0], invec, outvec, request);
-  CeedChk(ierr);
-  // Add to outvec with subsequent outputs
-  for (CeedInt i=1; i<numsub; i++) {
-    ierr = CeedOperatorGetData(suboperators[i], (void *)&impl); CeedChk(ierr);
-    impl->add = true;
-    ierr = CeedOperatorApply(suboperators[i], invec, outvec, request);
+  for (CeedInt i=0; i<numsub; i++) {
```

Comment:
@jedbrown, what do you thing about lifting this default implementation of applying a composite operator to the interface level now that we are adding ApplyAdd to all backends?

Reply:
Seems okay, but we want backends to be able to flatten the iteration space for composite operators. That could involve putting them all into a CUDA stream (processed sequentially by the GPU, but amortizes kernel launch/synchronization latency) or creating special kernels in which different element types are processed in different thread blocks (for example).

Reply:
I put the default impl in the interface but left the option to override. I should add a composite ApplyAdd test, but otherwise the codecov drop will be from not having any backends have separate impl for Apply vs ApplyAdd or having a backend impl of composite operators.